packages/lexical-react/src/LexicalTypeaheadMenuPlugin.tsx

```diff
@@ -297,6 +303,22 @@ export function LexicalTypeaheadMenuPlugin<TOption extends MenuOption>({
     openTypeahead,
   ]);
 
+  // Add effect to listen for editable state changes
+  useEffect(() => {
+    const editableListener = (isEditable: boolean) => {
+      if (!isEditable) {
+        closeTypeahead();
+      }
+    };
+
+    const removeEditableListener =
+      editor.registerEditableListener(editableListener);
+
+    return () => {
+      removeEditableListener();
+    };
```

Comment:
Just a nit for future reference, but this could be simpler:

```suggestion
    return editor.registerEditableListener(editableListener);
```

Reply:
@etrepum Thanks for the nit suggestion about simplifying the editable listener cleanup! I'll keep this in mind for future PRs. Sorry about the force push - I should have left the original approved version. Would you mind re-approving when you get a chance?

Reply:
Well if you're here, might as well make that simplification and I'll re-approve and add it to the merge queue once the CI test run finishes :)

Reply:
@etrepum I've pushed the simplified editable listener change using the implicit return approach. The change removes the intermediate variables and explicit return statement, making the code more concise while maintaining the same functionality. Looking forward to your re-review. Thanks  a lot for the feedback! 👍